### PR TITLE
Sources simplified

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: mule-sdk/v/4.0/1.5.1_sources_parameters.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
+[[_source_parameters]]
+= Defining parameters in a Message Source
+:keywords: mule, sdk, sources, listener, triggers, parameters
+
+
+== Basic sources parameters
+
+Parameters are defined as fields of your source class. All the features and annotations 
+available when defining an operation such as `@NullSafe`, `@ParameterGroup`, `@TypeResolver`,
```
Comment (pablomusumeci) [inline mule-sdk/v/4.0/1.5.1_sources_parameters.adoc]: Add references

File: mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* Throw an exception with a meaningful message
+
+If the source fails to start, so does the application that contains it.
+
+[TIP]
+All of the source parameters values that were defined through the `@Parameter` annotation are already resolved and available at this point.
+
+=== SourceCallback
+
+The method receives a SourceCallback object. The SourceCAllback is arguably the most important piece of the Source, since it is
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: `Source Callback`

Comment (juandesi) [reply mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: x2

File: mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+If the source fails to start, so does the application that contains it.
+
+[TIP]
+All of the source parameters values that were defined through the `@Parameter` annotation are already resolved and available at this point.
+
+=== SourceCallback
+
+The method receives a SourceCallback object. The SourceCAllback is arguably the most important piece of the Source, since it is 
+involved in pretty much every aspect of the source's functionality. We will elaborate on the uses and responsibilities of the SourceCallback
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: 3

File: mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+If the source fails to start, so does the application that contains it.
+
+[TIP]
+All of the source parameters values that were defined through the `@Parameter` annotation are already resolved and available at this point.
+
+=== SourceCallback
+
+The method receives a SourceCallback object. The SourceCAllback is arguably the most important piece of the Source, since it is 
+involved in pretty much every aspect of the source's functionality. We will elaborate on the uses and responsibilities of the SourceCallback 
+as this documentation progresses, but in a nutshell, the SourceCallback is what the source uses to communicate with the runtime. It is used for:
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: 4

File: mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+* Pushing messages into the Flow
+* Communicating connection issues and triggering reconnection
+* Passing state between the push phase of the Source into the response handling part
+
+Because the SourceCallback plays such an important role, it is very important that the provided instance is always available. Depending on how your source's
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: 5

File: mule-sdk/v/4.0/1.5.1_sources_parameters.adoc (hunk 1 above)
Comment (marianogonzalez) [reply mule-sdk/v/4.0/1.5.1_sources_parameters.adoc]: can't add references to things don't exist anymore. We'll do the referencing later

File: mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+:keywords: mule, sdk, sources, listener, triggers, lifecycle
+
+Unlike operations which lifecycle is tied to that of the owning Flow, sources can be started or stopped
+separately through the Runtime manager.
+
+To support that, The `Source` class forces you to implement two methods, `onStart()` and `onStop`
+
+== On Start
+
+Let's look at an example from the VM connector to see how the onStart method actually looks like:
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.2_sources_lifecycle.adoc]: `onStart()`

File: mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+ private HttpListenerConfig config;
+----
+
+== Obtaining a Connection
+
+If the source requires a connection (99% of connectors will, unless you’re implementing a scheduler), then you need to obtain 
+a connection. Just like with operations, the `@Connection` annotation is used for that, but with two big differences:
+
+* It is used in a field instead of a method argument
+* Instead of the connection object, a ConnectionProvider will be injected
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc]: `Conn Provider`

File: mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public void onStop() {
+        if (httpServer != null) { // <4>
+            serverProvider.disconnect(httpServer); // <5>
+        }
+    }
+}
+----
+
+<1>  We use the ConnectionProvider to obtain an HttpServer. The connection provider has most likely already
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc]: `ConnectionProvider`

File: mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+}
+----
+
+<1>  We use the ConnectionProvider to obtain an HttpServer. The connection provider has most likely already 
+configured the server with the proper host and port.
+<2> Http requests are pushed into the server by the remote clients and the  `HttpServer` component will
+notify the source of every request. The source then uses the sourceCallback to push the message into the flow
+<3> The processRequest method is one the Source should define to transform the http request into a message that
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc]: `Source`

Comment (juandesi) [reply mule-sdk/v/4.0/1.5.3_sources_config_connection.adoc]: sourceCallback -> `Source Callback`

File: mule-sdk/v/4.0/1.5.4_sources_push_message.adoc [hunk 10]
```diff
@@ -0,0 +1,42 @@
+[[_source_push_message]]
+= Pushing Messages to a Flow form a Message Source
+:keywords: mule, sdk, sources, listener, triggers, generate, push, message
+
+Once the source generates a piece of information that needs to be sent to the flow, it will do it through 
+the SourceCallback. Going back to the minimalistic Http listener example:
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.4_sources_push_message.adoc]: ` `

File: mule-sdk/v/4.0/1.5.4_sources_push_message.adoc [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+            sourceCallback.handle(result); // <4>
+        }); 
+    }
+}
+----
+
+<1>  We use the ConnectionProvider to obtain an HttpServer. The connection provider has most likely already 
+configured the server with the proper host and port.
+<2> Http requests are pushed into the server by the remote clients and the  `HttpServer` component will
+notify the source of every request. The source then uses the sourceCallback to push the message into the flow
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5.4_sources_push_message.adoc]: sourceCallback -> SourceCallback

File: mule-sdk/v/4.0/1.5.1_sources_parameters.adoc (hunk 1 above)
Comment (marianogonzalez) [reply mule-sdk/v/4.0/1.5.1_sources_parameters.adoc]: added a todo

File: mule-sdk/v/4.0/1.5_sources.adoc [hunk 12]
```diff
@@ -1,2 +1,61 @@
...
+
+[source, java]
+----
+@Alias("listener")
+@EmitsResponse
+public class HttpListener extends Source<InputStream, HttpRequestAttributes> {
+----
+
+[TIP]
+You can use the @Alias annotation to force a name, otherwise the SDK will infer one as usual
```
Comment (juandesi) [inline mule-sdk/v/4.0/1.5_sources.adoc]: ` alias`

**Pull request merged**